segmenters/nlp/SpacySentencizer/__init__.py

```diff
@@ -0,0 +1,94 @@
+from typing import Dict, List
+
+from jina.executors.segmenters import BaseSegmenter
+
+
+class SpacySentencizer(BaseSegmenter):
+    """
+    :class:`SpacySentencizer` split the text on the doc-level
+    into sentences on the chunk-level with spaCy as the backend.
+    The splitting technique is divided into two different approaches:
+    1) Default Segmenter: Utilizes dependency parsing to determine the rule for
+        splitting the text. For more info please refer to https://spacy.io/api/sentencizer.
+    2) Machine Learning-based Segmenter: Utilizes SentenceRecognizer model trained
+        on a dedicated data created for sentence segmentation. For more info please
+        refer to https://spacy.io/api/sentencerecognizer.
+
+    :param lang: pre-trained spaCy language pipeline,
+        by default "xx_sent_ud_sm".
+    :param use_default_segmenter: boolean to determine which segmentation algorithm to use,
+        by default False.
+    :param args:  Additional positional arguments
+    :param kwargs: Additional keyword arguments
+    """
+
+    SPACY_COMPONENTS = [
+        "tagger",
+        "parser",
+        "ner",
+        "senter",
+        "entity_linker",
+        "textcat",
+        "entity_ruler",
+        "merge_noun_chunks",
+        "merge_entities",
+        "merge_subtokens",
+        "lemmatizer",
+        "attribute_ruler",
+    ]
+
+    def __init__(self, lang: str = "xx_sent_ud_sm", use_default_segmenter: bool = False, *args, **kwargs):
+        """Set constructor."""
+        super().__init__(*args, **kwargs)
+
+        import spacy
+
+        try:
+            self.spacy_model = spacy.load(lang)
```

Comment:
All these expensive loads must be done in post_init().

Like this we ensure that expensive objects are not serialized by Jina